code/game/verbs/ooc.dm

```diff
@@ -14,6 +14,10 @@
 	msg = copytext(sanitize(msg), 1, MAX_MESSAGE_LEN)
 	if(!msg)	return
 
+	if((copytext(msg, 1, 2) in list(".",";",":","#")) || (lowertext(copytext(msg, 1, 4)) == "say"))
+		if(alert("Your message \"[msg]\" looks like it was meant for in game communication, say it in OOC?", "Meant for OOC?", "No", "Yes") != "Yes")

```

Comment:
OOC is the same for everyone, but say is pretty dependent on what you are. I'm afraid.